draft-ietf-quic-tls.md

```diff
@@ -1570,12 +1564,14 @@ Note:
   packet protection keys.  To avoid exhaustion of both sets of keys, it might be
   necessary to initiate two key updates in succession.
 
-For AEAD_AES_128_GCM, AEAD_AES_256_GCM, and AEAD_CHACHA20_POLY1305, the limit on
-the number of packets that fail authentication is 2^36.  Note that the analysis
-in {{AEBounds}} supports a higher limit for the AEAD_AES_128_GCM and
-AEAD_AES_256_GCM, but this specification recommends a lower limit.  For
-AEAD_AES_128_CCM, the limit on the number of packets that fail authentication
-is 2^23.5; see {{ccm-bounds}}.
+Key updates MUST be initiated before usage limits on packet protection keys are
+exceeded. For AEAD_AES_128_GCM and AEAD_AES_256_GCM, the confidentiality limit
+is 2^27 encrypted packets, and the integrity limit is 2^54 encrypted or forged
+packets; see {{?GCM-MU}} and {{gcm-bounds}}. For AEAD_CHACHA20_POLY1305, the
+limit on number of packets that fail authentication is 2^93; see {{AEBounds}}.
+For AEAD_AES_128_CCM, the confidentiality limit is 2^23 encrypted packets, and
+the integrity limit is 2^23.5 forged packets; see {{ccm-bounds}}. Future
+analyses and specifications MAY relax these limits.
```

Comment:
We probably need to split this paragraph now.  One for the text on key updates: update before you get to L[confidentiality].  And a separate one for stop using the connection when you get to L[integrity].

Reply:
How about:

```suggestion
Key updates MUST be initiated before confidentiality limits for the selected
AEAD are exceeded.  For AEAD_AES_128_GCM and AEAD_AES_256_GCM, the
confidentiality limit is 2^27 encrypted packets; see {{gcm-bounds}}.  For
AEAD_CHACHA20_POLY1305, the confidentiality limit is greater than the number of
possible packets (2^62) and so can be disregarded.  For AEAD_AES_128_CCM, the
confidentiality limit is 2^23 encrypted packets; see {{ccm-bounds}}.

A connection MUST be closed if the number of packets that fail authentication
exceed the integrity limit for the selected AEAD. For AEAD_AES_128_GCM, and the
integrity limit is 2^54 encrypted or forged packets; see {{gcm-bounds}}.  For
AEAD_AES_256_GCM, the integrity limit For AEAD_CHACHA20_POLY1305, the limit on
number of packets that fail authentication is 2^36; see {{AEBounds}}.  For
AEAD_AES_128_CCM, the integrity limit is 2^23.5 forged packets; see
{{ccm-bounds}}.

Future analyses and specifications MAY relax confidentiality or integrity limits
for an AEAD.
```

I changed the 2^93 to 2^36 for ChaCha.  I think that the 93 was in error.